python/unblob/models.py

```diff
@@ -30,6 +31,41 @@
 #
 
 
+class HandlerType(Enum):
+    ARCHIVE = "Archive"
+    COMPRESSION = "Compression"
+    FILESYSTEM = "FileSystem"
+    EXECUTABLE = "Executable"
+
+
+@dataclasses.dataclass(frozen=True)
+class Reference:
+    title: str
+    url: str
+
+
+@dataclasses.dataclass
+class HandlerDoc:
+    name: str
+    description: Union[str, None]
+    vendor: Union[str, None]
+    references: list[Reference]
+    limitations: list[str]
+    type: str = dataclasses.field(init=False)
+    # so HandlerDoc(type="whatever") is invalid construct, but HandlerDoc.type will exist in the end
+    handler_type: dataclasses.InitVar[HandlerType]
+    # HandlerDoc(handler_type=HandlerType...) will work, but HandlerDoc.handler_type will not exist
+    # it will be passed to __post_init__ as argument
+    fully_supported: bool = dataclasses.field(default=True)
+    # there is a default, but can also be overridden to False
+    # also it is set to False, if there are limitations
+    # alternatively, it can also be a field(init=False), and set only in __post_init__
+
+    def __post_init__(self, handler_type):
+        self.type = handler_type.name
+        self.fully_supported = self.fully_supported and not self.limitations
```

Comment:
Well, these comments were not intended to land in the code :)

Seeing the [unintended] change of handler type being capitalized in the current version, I would also revise/simplify it one bit further: actually we could store the `HandlerType` value in the `handler_type` field and remove the `type` field, eliminating also the `InitVar`.

```py
    limitations: list[str]
    handler_type: HandlerType
...
   def __post_init__(self):
       self.fully_supported = self.fully_supported and not self.limitations
```

This would also need a follow up in `generate_markdown()`

Reply:
Done.